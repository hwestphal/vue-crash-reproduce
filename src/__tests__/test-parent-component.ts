import ParentComponent from "../parent-component.vue";
import SomeComponent from "../some-component.vue";
import { createLocalVue, mount } from "@vue/test-utils";

// This unit test is failing.
test("parent component", () => {
    const localVue = createLocalVue();
    localVue.mixin({
        beforeCreate() {
            (this as any).$content = "some value";
        },
    });
    const wrapper = mount(ParentComponent, { localVue });
    expect((wrapper.vm as any).$content).not.toBeUndefined();
    expect((wrapper.find(SomeComponent).vm as any).$content).not.toBeUndefined();
    expect(wrapper.element).toMatchSnapshot();
});
