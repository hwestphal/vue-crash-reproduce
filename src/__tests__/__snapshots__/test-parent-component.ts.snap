// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parent component 1`] = `
<div>
  <div>
    <p>
      SOME VALUE
    </p>
  </div>
</div>
`;
